packages/server/src/admin/invite.ts

```diff
@@ -253,13 +254,87 @@ async function upsertProfileResource(
   }
 }
 
+/**
+ * Validates that all access policy references exist and belong to the project.
+ * @param systemRepo - The system repository.
+ * @param request - The invite request containing access policy references.
+ * @param project - The project to validate against.
+ * @throws OperationOutcomeError if any access policy is invalid.
+ */
+async function validateAccessPolicies(
+  systemRepo: Repository,
+  request: ServerInviteRequest,
+  project: WithId<Project>
+): Promise<void> {
+  // Validate single accessPolicy if provided
+  if (request.accessPolicy) {
+    await validateAccessPolicyReference(systemRepo, request.accessPolicy, project);
+  }
+
+  // Validate access array if provided
+  if (request.access && Array.isArray(request.access)) {
+    for (const access of request.access) {
+      if (access.policy) {
+        await validateAccessPolicyReference(systemRepo, access.policy, project);
+      }
+    }
+  }
+
+  // Validate membership.access array if provided
+  if (request.membership?.access && Array.isArray(request.membership.access)) {
+    for (const access of request.membership.access) {
+      if (access.policy) {
+        await validateAccessPolicyReference(systemRepo, access.policy, project);
```

Comment:
Could this instead gather up all the references and make a single `readReferences()` call to check them all at once?